### PR TITLE
[added] recursive discovery of varz values

### PR DESCRIPTION
**Pull request by matthiashanel**
Signed-off-by: Matthias Hanel <mh@synadia.com>

```
# HELP gnatsd_varz_jetstream_config_domain jetstream_config_domain
# TYPE gnatsd_varz_jetstream_config_domain gauge
gnatsd_varz_jetstream_config_domain{server_id="http://localhost:8080",value="hub"} 1
# HELP gnatsd_varz_jetstream_config_max_memory jetstream_config_max_memory
# TYPE gnatsd_varz_jetstream_config_max_memory gauge
gnatsd_varz_jetstream_config_max_memory{server_id="http://localhost:8080"} 5.1539607552e+10
# HELP gnatsd_varz_jetstream_config_max_storage jetstream_config_max_storage
# TYPE gnatsd_varz_jetstream_config_max_storage gauge
gnatsd_varz_jetstream_config_max_storage{server_id="http://localhost:8080"} 5.35063858176e+11
# HELP gnatsd_varz_jetstream_meta_leader jetstream_meta_leader
# TYPE gnatsd_varz_jetstream_meta_leader gauge
gnatsd_varz_jetstream_meta_leader{server_id="http://localhost:8080",value="srv-4282"} 1
# HELP gnatsd_varz_jetstream_stats_accounts jetstream_stats_accounts
# TYPE gnatsd_varz_jetstream_stats_accounts gauge
gnatsd_varz_jetstream_stats_accounts{server_id="http://localhost:8080"} 1
# HELP gnatsd_varz_jetstream_stats_api_errors jetstream_stats_api_errors
# TYPE gnatsd_varz_jetstream_stats_api_errors gauge
gnatsd_varz_jetstream_stats_api_errors{server_id="http://localhost:8080"} 2
# HELP gnatsd_varz_jetstream_stats_api_total jetstream_stats_api_total
# TYPE gnatsd_varz_jetstream_stats_api_total gauge
gnatsd_varz_jetstream_stats_api_total{server_id="http://localhost:8080"} 0
# HELP gnatsd_varz_jetstream_stats_memory jetstream_stats_memory
# TYPE gnatsd_varz_jetstream_stats_memory gauge
gnatsd_varz_jetstream_stats_memory{server_id="http://localhost:8080"} 0
# HELP gnatsd_varz_jetstream_stats_storage jetstream_stats_storage
# TYPE gnatsd_varz_jetstream_stats_storage gauge
gnatsd_varz_jetstream_stats_storage{server_id="http://localhost:8080"} 0
```

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -275,30 +292,74 @@ func (nc *NATSCollector) initMetricsFromServers(namespace string) {
 		}
 	}
 
+	nc.objectToMetrics(response, namespace)
+}
+
+// returns a sanitized fully qualified name and path
+func fqNane(name string, prefix ...string) (string, []string) {
```
Comment (variadico) [inline collector/collector.go:299]: Possible typo.
```suggestion
func fqName(name string, prefix ...string) (string, []string) {
```

File: collector/collector.go [hunk 2]
```diff
@@ -275,30 +292,74 @@ func (nc *NATSCollector) initMetricsFromServers(namespace string) {
...
-				nc.Stats[k] = newPrometheusGaugeVec(nc.system, nc.endpoint, k, "", namespace)
-			case string:
-				if _, ok := labelKeys[k]; !ok {
-					break
-				}
-				nc.Stats[k] = newLabelGauge(nc.system, nc.endpoint, k, "", namespace, "value")
-			default:
-				// not one of the types currently handled
-				Tracef("Unknown type:  %v, %v", k, v)
+		fqn, path := fqNane(k, prefix...)
```
Comment (variadico) [inline collector/collector.go:334]: Possible typo.
```suggestion
		fqn, path := fqName(k, prefix...)
```

**Comment by matthiashanel** [comment]
@variadico Thanks for finding

**Comment by variadico** [comment]
If you rebase against latest master, the CI error you're seeing will be fixed via https://github.com/nats-io/prometheus-nats-exporter/pull/140.

**Review by wallyqs — APPROVED** [review]
LGTM

**Pull request merged**